test/WebJobs.Script.Tests/ManagedDependencies/ManagedDependencyProviderTests.cs

```diff
@@ -0,0 +1,263 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+using Microsoft.Azure.WebJobs.Script.Eventing;
+using Microsoft.Azure.WebJobs.Script.ManagedDependencies;
+using Microsoft.Azure.WebJobs.Script.Rpc;
+using Microsoft.Extensions.Options;
+using Moq;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.ManagedDependencies
+{
+    public class ManagedDependencyProviderTests
+    {
+        private readonly IManagedDependencyProvider managedDependencyProvider;
+        private readonly OSPlatform osPlatform = OSPlatform.Windows;
+        private readonly IManagedDependencyConfigs testManagedDependencyConfigs;
+
+        public ManagedDependencyProviderTests()
+        {
+            managedDependencyProvider = new ManagedDependencyProvider();
+            osPlatform = OSPlatform.Windows;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
```

Comment:
Same comments here use IEnvironment and write tests that mock both windows and linux